addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js

```diff
@@ -37,6 +38,37 @@ odoo.define('point_of_sale.ProductsWidgetControlPanel', function(require) {
             this.trigger('switch-category', 0);
             this.trigger('update-search', productName);
         }
+        async loadProductFromDB() {
+            if(!this.searchWordInput.el.value)
+                return;
+
+            try {
+                let ProductIds = await this.rpc({
+                    model: 'product.product',
+                    method: 'search',
+                    args: [[['name', 'ilike', this.searchWordInput.el.value + "%"]]],
+                    context: this.env.session.user_context,
+                });
+                if(!ProductIds.length) {
+                    this.showPopup('ErrorPopup', {
+                        title: this.env._t(''),
+                        body: this.env._t("No product found"),
+                    });
+                } else {
+                    await this.env.pos._addProducts(ProductIds);
+                }
+                this.trigger('update-product-list');
+            } catch (error) {
+                if (identifyError(error) && error.message.code < 0) {
```

Comment:
`identifyError` returns an error by which you need to check the type. You should probably do:
```js
const identifiedError = identifyError(error)
if (identifiedError instanceof ConnectionLostError || identifiedError instanceof ConnectionAbortedError) {
   ...
}
```

You will need to import ConnectionLostError and ConnectionAbortedError like so:

```js
    const { ConnectionLostError, ConnectionAbortedError } = require('@web/core/network/rpc_service');
```